capz/run-capz-e2e.sh

```diff
@@ -373,13 +381,13 @@ run_e2e_test() {
         fi
 
         ADDITIONAL_E2E_ARGS=()
-        if [[ "$CI" == "true" ]]; then
-            # private image repository doesn't have a way to promote images: https://github.com/kubernetes/k8s.io/pull/1929
-            # So we are using a custom repository for the test "Container Runtime blackbox test when running a container with a new image should be able to pull from private registry with secret [NodeConformance]"
-            # Must also set label preset-windows-private-registry-cred: "true" on the job
-            export KUBE_TEST_REPO_LIST="$SCRIPT_ROOT/../images/image-repo-list-private-registry"
-            ADDITIONAL_E2E_ARGS+=("--docker-config-file=${DOCKER_CONFIG_FILE}")
```

Comment:
This will not be in the community sub, we will move this to a private registry in the private sub and use kubelet credential provider to access it.  we need to wire this in to our templates but the credential binaries are already there https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/4679#issuecomment-2225938449

Reply:
https://github.com/kubernetes-sigs/windows-testing/issues/446